pagopa/io-app
# [IOCOM-1404, IOCOM-1405] Empty list and error component

## Short description
This PR adds the empty list component to the message's home, new DS.

|Inbox<br/>No data|Archive<br/>No data|Error<br/>(both)|
|-|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-06-03 at 16 36 16](https://github.com/pagopa/io-app/assets/5150343/70203e33-3e54-4ecb-91f9-4e9b8b219a4c)|![Simulator Screenshot - iPhone 15 - 2024-06-03 at 16 36 21](https://github.com/pagopa/io-app/assets/5150343/16191d83-2340-4ac6-940c-7f2e6f416ef8)|![Simulator Screenshot - iPhone 15 - 2024-06-03 at 16 34 44](https://github.com/pagopa/io-app/assets/5150343/ef3d04d8-ab84-408c-a58c-ad13bd2a7bad)|

## List of changes proposed in this pull request
- `EmptyList` component. Given an input `category` property (`inbox` or `archive`), it uses a selector to know if there is an error or there are no data. It also has a fallback to render nothing (although it never happens, since that is the case where the list has data and the component is not rendered at all)
- `emptyListReasonSelector` selector which output is:
  - `noData` when no messages have been retried yet or they have been retrieved as an empty list
  - `error` when the retrieval failed, without any successful previous retrieval (the case where there is an error with available data will be implemented later, using a toast, in a future PR)
  - `notEmpty` maps all cases where the list has valid data to show (and the `EmptyList` component is hidden)


## How to test
Using the io-dev-api-server, set the messages' configuration to generate zero messages (use a delay of 3500ms to see the transition between loading and empty). In order to see the error, set the `getMessagesResponseCode` to 500.
